server/middleware/authHandlerMiddleware.js

```diff
@@ -1,13 +1,14 @@
 const userAuthenticationService = require('../services/userAuthenticationService')
 
-module.exports = async function authHandler(req, res, next) {
+const authHandler = async (req, res, next) => {
   const userSessionExpiryDateTime = await userAuthenticationService.getSessionExpiryDateTime(req.user.username)
-
-  if (userSessionExpiryDateTime !== null && userSessionExpiryDateTime[0] != null) {
-    if (new Date() > new Date(userSessionExpiryDateTime[0].SessionExpiryDateTime)) {
-      res.redirect('/logout')
-    } else {
-      next()
-    }
-  }
+  if (
+    !userSessionExpiryDateTime ||
+    !userSessionExpiryDateTime[0] ||
+    new Date() > new Date(userSessionExpiryDateTime[0].SessionExpiryDateTime)
+  )
+    return res.redirect('/auth/login')
+  return next()
```

Comment:
I think I've learnt the answer by now but I'm ready to be surprised. Is there any existing coverage for this?

Reply:
and would it still not be three &&'s